File: backend/agentic_crm_sdk/sequences.py
"""
Sequences API client module.
"""
from agentic_crm_sdk.client import AgenticCRM


class SequencesClient:
    def __init__(self, client: AgenticCRM):
        self._client = client

    def list(self, **params):
        return self._client.get("/api/sequences", params=params)

    def get(self, sequence_id: int):
        return self._client.get(f"/api/sequences/{sequence_id}")

    def create(self, data: dict):
        return self._client.post("/api/sequences", json=data)

    def update(self, sequence_id: int, data: dict):
        return self._client.patch(f"/api/sequences/{sequence_id}", json=data)

    def delete(self, sequence_id: int):
        return self._client.delete(f"/api/sequences/{sequence_id}")

    def enroll_lead(self, sequence_id: int, lead_id: int):
        return self._client.post(f"/api/sequences/{sequence_id}/enroll", json={"lead_id": lead_id})

File: packages/js-sdk/src/sequences.ts
import { AgenticCRM } from './client'

export interface SequenceStep {
  type: string
  subject?: string
  content: string
  delay_days: number
}

export interface Sequence {
  id: number
  name: string
  description?: string
  steps: SequenceStep[]
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface CreateSequenceInput {
  name: string
  description?: string
  steps: SequenceStep[]
  is_active?: boolean
}

export interface UpdateSequenceInput {
  name?: string
  description?: string
  steps?: SequenceStep[]
  is_active?: boolean
}

export class SequencesClient {
  constructor(private client: AgenticCRM) {}

  async list(): Promise<Sequence[]> {
    return this.client.get<Sequence[]>('/api/sequences')
  }

  async get(sequenceId: number): Promise<Sequence> {
    return this.client.get<Sequence>(`/api/sequences/${sequenceId}`)
  }

  async create(data: CreateSequenceInput): Promise<Sequence> {
    return this.client.post<Sequence>('/api/sequences', data)
  }

  async update(sequenceId: number, data: UpdateSequenceInput): Promise<Sequence> {
    return this.client.patch<Sequence>(`/api/sequences/${sequenceId}`, data)
  }
}